fg_lib/include/fg_lib.h

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @file
+ * @brief  Nordic Fuel Gauge library API.
+ */
+
+#ifndef __FG_LIB_H__
+#define __FG_LIB_H__
+
+/**
+ * @defgroup fg_lib Battery fuel gauge library
+ * @{
+ *
+ * @brief Algorithm for battery state-of-charge estimation ("fuel gauge").
+ *
+ * @details The algorithm determines the predicted state-of-charge (SOC) based on the following
+ * data: model of the specific battery type used in the given application along with
+ * measurements of battery voltage, current, and temperature.
+ * The SOC is a percentage value between 0% and 100%.
+ *
+ * @note Current and voltage measurements do not have to be made at a certain interval.
+ *       However, the two values should be measured as closely as possible in time.
+ */
+
+#include <stddef.h>
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Version number of battery fuel gauge library in format "x.y.z".
+ */
+extern const char *fg_lib_version;
+
+/**
+ * @brief Build date of the battery fuel gauge library.
+ */
+extern const char *fg_lib_build_date;
+
+/**
+ * @brief Battery model parameters.
+ *
+ * @details Parameter details purposefully not documented.
+ */
+struct battery_model {
+	float param_1[201];
+	float temps[3];
+	float param_2[122];
+	float param_3[201];
+	float param_4[201];
+	float param_5[201];
+	float param_6[201];
+	float param_7[122];
+	float param_8[122];
+	float param_9[3];
+	float param_10[6];
+	float param_11[6];
+	float param_12[3];
+	char name[64];
+};
+
+/**
+ * @brief Library state information. Useful for debugging.
+ */
+struct fg_lib_state_info {
+	/** Predicted voltage. */
+	float yhat;
+	/** Battery resistance. */
+	float r0;
+	/** Truncated temperature. This state is used when the measured temperature falls outside
+	 * the modeled range.
+	 */
+	float T_truncated;
+};
+
+/**
+ * @brief Library init parameter struct.
+ */
+struct fg_lib_init_parameters {
+	/** Initial voltage in volts [V]. */
+	float v0;
+	/** Initial current in amperes [A]. */
+	float i0;
+	/** Initial temperature in centigrades [C]. */
+	float t0;
+	/** Battery model */
```

Comment:
```suggestion
	/** Battery model. */
```

Reply:
Maybe add an example here?

Reply:
The model structs are quite large and would clutter the documentation, I think. We are working on adding samples in sdk-nrf that will contain example battery models. Will that be sufficient?